simplejson/_speedups.c

```diff
@@ -2953,6 +2956,9 @@ encoder_listencode_dict(PyEncoderObject *s, JSON_Accu *rval, PyObject *dct, Py_s
     PyObject *encoded = NULL;
     Py_ssize_t idx;
 
+    if (!PyDict_Check(dct)) {
+        return -1;
+    }
```

Comment:
This should be done at the call site in here, so that a proper exception can be raised and a redundant check is not done in the dictionary branch below

https://github.com/simplejson/simplejson/blob/5a40364f951f85ed73d6aaf881f8ee9521a56f07/simplejson/_speedups.c#L2850-L2860